[Mondrian] Add viewport dynamic limiter bar

ai-moiseev opened the pull request:
This commit adds viewport limiter bar which provides ability to set min/max cycle values to avoid hang-ups for a very large models.

Areas outside of active 'viewport' are marked with shadow borders.

Issue: #395 

Screenshot (shadow on left and right marks viewport limit):
![image](https://user-images.githubusercontent.com/67966333/171845911-da42f2fe-d29b-4bc4-8db7-8d52b93aa18f.png)

Signed-off-by: Alexander Moiseev <a.moiseev@samsung.com>

ai-moiseev commented:
@hyunsik-yoon @llFreetimell I don't have a membership in this repository and can't add reviewers, PTAL :)

hyunsik-yoon commented:
@ai-moiseev 
I tried to run this with `sample.tracealloc.json`. I saw a big scroll bar on the top. I tried to click `<` and `>` button and area with `cycle` text but nothing happened.

Could you explain what this scroll bar means to users? If I need some other sample `tracealloc.json` to see the necessity of this scroll bar, could you attach the json file? Thanks.

ai-moiseev commented:
@hyunsik-yoon This PR adds functionality from original `Mondrian` utility which allowed to render only selected cycle range (min and max were set as CLI arguments) to provide support for a very large models. I can send you some of larger JSON files via e-mail, but I'm not sure if we can upload it here as it may contain some sensitive information from production models. As an alternative I can generate JSON with fake data for a viewer performance testing.

I've replaced `<`/`>` arrows with `||` to better reflect that these buttons are mouse-draggable:

![image](https://user-images.githubusercontent.com/67966333/172416690-76d0556c-ce44-4505-9b4c-d3eac7c6cac3.png)

I also added note to `README` to describe what these UI elements should do.

llFreetimell commented on media/Mondrian/mondrianViewer.js:
```diff
@@ -28,6 +28,10 @@ const cycleCountContainer =
+const scrollbarBtn = /** @type {HTMLElement} */ document.querySelector('.mondrian-scrollbar-btn');
+
+/* Limit cycle range to avoid hang-ups for large models */
+const defaultCycleLimit = 1048576;
```
Just curious, why this value is used? :D

ai-moiseev commented on media/Mondrian/style.css:
```diff
@@ -27,6 +27,37 @@ div.mondrian-scrollbar {
+  min-width: 32px;
+  white-space: nowrap;
+  overflow: hidden;
+  color: var(--vscode-button-secondaryForeground);
```
Default style sheet defines a lot of variables we can use to adapt to current theme :)

Vscode has full-featured WebTools from Chrome (`Ctlr+Shift+P` >> `Open web tools`), so it's possible to check how various UI elements are implemented:

![1654704207](https://user-images.githubusercontent.com/67966333/172664489-10aaa564-d065-4351-9170-db9421663cad.png)

llFreetimell replied:
> Vscode has full-featured WebTools from Chrome

As I am not good at web programming, I didn't know that it has such wonderful features :open_mouth: 
Thanks for good information :)

llFreetimell reviewed (approved):
LGTM

ai-moiseev commented:
@llFreetimell Yes, I think it's OK :) I'm going on vacation myself for a week.
BTW, do you know if it's possible to get membership in this repository? It would be nice to be able add reviewers, sign up for tasks in tracking issues etc. :)

llFreetimell commented:
> it's possible to get a membership in this repository?

@lemmaa Is it possible? @.@

lemmaa commented:
> > it's possible to get a membership in this repository?
> 
> @lemmaa Is it possible? @.@

@llFreetimell , sure!

@ai-moiseev , sorry for missing you. You are now added as a member of the @Samsung/one-vscode team. Thank you always for your contribution. :)
